### ISSUE TITLE
Account Login Via API still redirects to customer force sign in page

### ISSUE THREAD

**Issue by michaelfigg** [bitExpert/magento2-force-login#142]
I am logging in customers through a third party portal. It uses the Magento API. Without activating the force login plugin customers can login to the website via the portal fine and will auto login the customer to the magento login page. However when I activate the plugin and my customer's login through the portal, they are redirected to the website but are forced to the login page again. I looked at the whitelist rules and added all the urls I think would be e.g the link that should work is /companyaccount/api/login?email=name%40domain.com&&token=tokennumber so I added the following rules to the whitelist:
/companyaccount
/companyaccount/api
/companyaccount/api/login
even: /customer/account/login/referer

but no luck, 

Im using magento 2.2.5

does this plugin support the magento api for customers to login and if not how can we get this to work

Thanks in advance

**Comment by shochdoerfer** [comment]
I'd assume the api urls do not need to be whitelisted. I assume that the login via API works slightly different than via storefront and thus our module is not aware that the customer is properly logged in already. Try to debug in LoginCheck:execute() what does go wrong here. I assume the following check does not work as expected:
```
if ($this->customerSession instanceof \Magento\Customer\Model\Session &&
    $this->customerSession->isLoggedIn()) {
        return false;
}
```
I'd be happy to include a patch to make this work but won't find the time right now to debug & work on this. Help is appreciated.

**Comment by michaelfigg** [comment]
Hey shochdoerfer, 

Could I have your skype please so I could discuss with you about a patch please?

Thanks in advnce

**Comment by shochdoerfer** [comment]
@michaelfigg sure, my Skype nick is bitExpert. Can't be avail today though.

**Comment by shochdoerfer** [comment]
@michaelfigg don't forget to send in your PR ;)

**Comment by shochdoerfer** [comment]
ping @michaelfigg - still waiting for your PR. Would be great to have your use case supported by the module. What did you change to make it happen?

**Comment by shochdoerfer** [comment]
@michaelfigg any change you could provide us a PR or at least a pointer what you did to make it work for you?

**Comment by gewaechshaus** [comment]
@shochdoerfer  - Hallo Stephan,

we have the same problem... Any news related to this?

Viele Grüße
Jan

**Comment by gewaechshaus** [comment]
$this->_logger->info('BitExpert LoginCheck');
        $this->_logger->info($this->customerSession->isLoggedIn());
        $this->_logger->info($this->customerSession instanceof \Magento\Customer\Model\Session);

=
[2019-10-09 10:38:02] main.INFO: BitExpert LoginCheck [] []
[2019-10-09 10:38:02] main.INFO:  [] []
[2019-10-09 10:38:02] main.INFO: 1 [] []

**Comment by shochdoerfer** [comment]
@gewaechshaus did not yet find the time to dive into this. Happy to accept PR's if you found a good way to fix this.

**Comment by gewaechshaus** [comment]
@shochdoerfer - maybe it would make sense to dispatch a custom event?

**Comment by gewaechshaus** [comment]
I found a way to handle this with login as customer module.
I inserted at https://github.com/bitExpert/magento2-force-login/blob/5c09aa1b0ca9587aca41c2d94c2f02ec2b44bb41/Controller/LoginCheck.php#L160

       $afterLoginReferer = $this->session->getData()['after_login_referer'];

        if (strpos($afterLoginReferer, 'loginascustomer') !== false) {
            return false;
        }

This should work with every login as customer module using a frontend route.

**Comment by shochdoerfer** [comment]
Fixed Magento Admin login as customer with #212
